[CI] FunctionScoreQueryBuilderTests testCacheability failure

Failure on master: https://gradle-enterprise.elastic.co/s/dmut5fjhfhzh2

This reproduces locally with
```
./gradlew ':server:test' --tests "org.elasticsearch.index.query.functionscore.FunctionScoreQueryBuilderTests.testCacheability" \
  -Dtests.seed=E852D4E6B71A6977 \
  -Dtests.security.manager=true \
  -Dtests.locale=it \
  -Dtests.timezone=SST \
  -Dcompiler.java=14
```

Error:
```
java.lang.AssertionError: query should not be cacheable: {
  "function_score" : {
    "query" : {
      "term" : {
        "BCGkj" : {
          "value" : true,
          "boost" : 0.2
        }
      }
    },
    "functions" : [
      {
        "filter" : {
          "ids" : {
            "values" : [
              "IqbhWHRudT",
              "CY",
              "xsJoX",
              "I",
              "vYm",
              "QhctenW",
              "IrQzUnqvn",
              "m",
              "dpvkCJK"
            ],
            "boost" : 0.6666667,
            "_name" : "ohSBWZ160"
          }
        },
        "weight" : 0.92822886,
        "script_score" : {
          "script" : {
            "source" : "1",
            "lang" : "mockscript"
          }
        }
      }
    ],
    "score_mode" : "first",
    "max_boost" : 0.6829204,
    "boost" : 1.0,
    "_name" : "f161"
  }
} expected:<false> but was:<true>
	at __randomizedtesting.SeedInfo.seed([E852D4E6B71A6977:C8DBA06ABD824177]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.elasticsearch.index.query.functionscore.FunctionScoreQueryBuilderTests.testCacheability(FunctionScoreQueryBuilderTests.java:832)
```

--- a/server/src/test/java/org/elasticsearch/index/query/functionscore/FunctionScoreQueryBuilderTests.java
+++ b/server/src/test/java/org/elasticsearch/index/query/functionscore/FunctionScoreQueryBuilderTests.java
@@ -20,6 +20,7 @@
 package org.elasticsearch.index.query.functionscore;
 
 import com.fasterxml.jackson.core.JsonParseException;
+
 import org.apache.lucene.index.Term;
 import org.apache.lucene.search.MatchAllDocsQuery;
 import org.apache.lucene.search.MatchNoDocsQuery;
@@ -829,6 +830,10 @@ public class FunctionScoreQueryBuilderTests extends AbstractQueryTestCase<Functi
         QueryShardContext context = createShardContext();
         QueryBuilder rewriteQuery = rewriteQuery(queryBuilder, new QueryShardContext(context));
         assertNotNull(rewriteQuery.toQuery(context));
+        // we occasionally need to update the expected "isCacheable" flag after rewrite for MatchNoneQueryBuilder
+        if (rewriteQuery instanceof MatchNoneQueryBuilder) {
+            isCacheable = true;
+        }
         assertEquals("query should " + (isCacheable ? "" : "not") + " be cacheable: " + queryBuilder.toString(), isCacheable,
                 context.isCacheable());
 

